src/Azure.Functions.Cli/npm/lib/main.js

```diff
@@ -7,19 +7,12 @@ var fork = require('child_process').fork;
 var args = process.argv;
 
 function main() {
-    var isWin = /^win/.test(process.platform);
     var bin = path.join(path.dirname(fs.realpathSync(__filename)), '../bin');
-
-    if (!isWin) {
-        console.log('Currently Azure Functions Core Tools is Windows only.\n');
-        console.log('Follow https://github.com/Azure/azure-functions-cli/issues/178 for updates.');
-        process.exit(1);
-    } else {
-        var funcProc = spawn(bin + '/func.exe', args.slice(2), { stdio : [process.stdin, process.stdout, process.stderr, 'pipe']});
-        funcProc.on('exit', function(code) {
-            process.exit(code);
-        });
-    }
+    
+    var funcProc = spawn('dotnet', [bin + '/Azure.Functions.Cli.dll',].concat(args.slice(2)), { stdio : [process.stdin, process.stdout, process.stderr, 'pipe']});
```

Comment:
This is pretty naive, as it just assumes .NET Core is installed. We'd either need to check and provide instructions or make sure the CLI is self contained